lib/board.rb

```diff
@@ -0,0 +1,54 @@
+class Board
+
+  attr_accessor :cells
+
+  @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+
+  def initialize
+    reset!
+  end
+
+  def reset!
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    row0 = " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    row1 = " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    row2 = " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    lines = '-----------'
+    spaces = '           '
+    puts row0, lines, row1, lines, row2, spaces
+  end
+
+  def position(user_input)
+    num = user_input.to_i - 1
+      self.cells[num]
+  end
+
+  def full?
+    switch = true
+    self.cells.each {|item| item != " " ? switch : switch = false}
+    switch
+  end
+
+  def turn_count
+    count = 0
+    self.cells.each {|item| item != " " ? count += 1 : count}
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```